cmd/global-query/pkg/distributed/query.go

```diff
@@ -306,7 +304,7 @@ func finalizeResult(res *results.Result, stmt *query.Statement, rowMap results.R
 	// assign the rows to the result
 	res.Rows = rowMap.ToRowsSorted(results.By(stmt.SortBy, stmt.Direction, stmt.SortAscending))
 
-	limit := min(stmt.NumResults, maxLimitStreaming)
+	limit := min(stmt.NumResults, limitUpperBound)
```

Comment:
Before, `maxLimitStreaming` would also kick in for the final call to `finalizeResult`, leading to truncation all the time, SSE partialResult or not.